### PR TITLE
Add HttpSender interface to decouple rewrite-maven from OkHttp

### PR DESCRIPTION
**Pull request by jkschneider**
See #1511.

### CODE REVIEW

File: rewrite-core/src/main/java/org/openrewrite/ipc/http/HttpSender.java [hunk 1]
```diff
@@ -0,0 +1,403 @@
...
+
+            private static byte[] gzip(byte[] data) throws IOException {
+                ByteArrayOutputStream bos = new ByteArrayOutputStream(data.length);
+                try (GZIPOutputStream out = new GZIPOutputStream(bos)) {
+                    out.write(data);
+                }
+                return bos.toByteArray();
+            }
+
+            public final Builder print() {
```
Comment (tkvangorder) [inline rewrite-core/src/main/java/org/openrewrite/ipc/http/HttpSender.java:292]: Probably just change `print` to  `toString()` and leave it to the caller to decide if they want to print it?

**Review by sambsnyd — APPROVED** [review]

File: rewrite-maven/src/main/java/org/openrewrite/maven/MavenVisitor.java [hunk 2]
```diff
@@ -283,7 +284,7 @@ public ResolvedDependency findDependency(Xml.Tag tag, @Nullable Scope inClasspat
     }
 
     public MavenMetadata downloadMetadata(String groupId, String artifactId, ExecutionContext ctx) {
```
Comment (tkvangorder) [inline rewrite-maven/src/main/java/org/openrewrite/maven/MavenVisitor.java:286]: Wondering if the HttpSender should be something that can be configured in the context? Right now, any recipe using this is going to be using the default with no ability to change the implementation (not a big deal).

**Review by tkvangorder — APPROVED** [review]
Just a couple of minor suggestions.

File: rewrite-maven/src/main/java/org/openrewrite/maven/MavenVisitor.java (hunk 2 above)
Comment (jkschneider) [reply rewrite-maven/src/main/java/org/openrewrite/maven/MavenVisitor.java:286]: Thought about it, but I don't know how we would serialize that over the wire. So deferring to later.

Comment (tkvangorder) [reply rewrite-maven/src/main/java/org/openrewrite/maven/MavenVisitor.java:286]: oh good point.

**Pull request merged**